client/image_list_test.go

```diff
@@ -158,3 +159,66 @@ func TestImageListApiBefore125(t *testing.T) {
 		t.Fatalf("expected 2 images, got %v", images)
 	}
 }
+
+// Checks if shared-size query parameter is set/not being set correctly
+// for /images/json.
+func TestImageListWithSharedSize(t *testing.T) {
+	// performs ImageList call and returns the sent url query
+	getImageListAndGetQuery := func(version string, options types.ImageListOptions) (query url.Values, err error) {
+		client := &Client{
+			client: newMockClient(func(req *http.Request) (*http.Response, error) {
+				query = req.URL.Query()
+
+				return &http.Response{
+					StatusCode: http.StatusOK,
+					Body:       io.NopCloser(strings.NewReader("[]")),
+				}, nil
+			}),
+			version: version,
+		}
+
+		_, err = client.ImageList(context.Background(), options)
+		return query, err
+	}
+
+	const sharedSize = "shared-size"
+
+	// Check if the shared-size query parameter is unset
+	for _, tC := range []struct {
+		name    string
+		version string
+		options types.ImageListOptions
+	}{
+		{name: "after 1.42, if not requested", version: "1.42", options: types.ImageListOptions{SharedSize: false}},
+		{name: "before 1.42, even if requested", version: "1.41", options: types.ImageListOptions{SharedSize: true}},
+	} {
+		t.Run("unset "+tC.name, func(t *testing.T) {
```

Comment:
We should probably capture `tc` here (just in case things are run in parallel); perhaps we should actually enable `t.Parallel()` (we should do that more often).

```suggestion
	} {
		tc := tc	
		t.Run("unset "+tc.name, func(t *testing.T) {
```

Reply:
Thanks, the test cases amount is small, so I thought it's not worth to make it `t.Parallel()`. Can change it though, it doesn't hurt I guess.